undertow/src/main/java/org/wildfly/extension/undertow/RemoteHttpInvokerService.java

```diff
@@ -22,38 +22,49 @@
 
 package org.wildfly.extension.undertow;
 
+import io.undertow.server.HttpHandler;
 import io.undertow.server.handlers.CookieImpl;
 import io.undertow.server.session.SecureRandomSessionIdGenerator;
 import org.jboss.msc.service.Service;
 import org.jboss.msc.service.StartContext;
 import org.jboss.msc.service.StartException;
 import org.jboss.msc.service.StopContext;
 import io.undertow.server.handlers.PathHandler;
+import org.wildfly.httpclient.common.HttpServiceConfig;
 
 /**
- * Service that exposes the Wildfly
+ * Service that exposes the Wildfly Remote HTTP invoker base PathHandler.
  *
  * @author Stuart Douglas
+ * @author Richard Achmatowicz
  */
 public class RemoteHttpInvokerService implements Service<PathHandler> {
 
-    private static final String AFFINITY_PATH = "/common/v1/affinity";
+    private static final String COMMON_PATH = "/common";
+    private static final String AFFINITY_PATH = "/affinity";
 
     private final PathHandler pathHandler = new PathHandler();
 
     @Override
     public void start(StartContext context) throws StartException {
         pathHandler.clearPaths();
-        SecureRandomSessionIdGenerator generator = new SecureRandomSessionIdGenerator();
 
-        pathHandler.addPrefixPath(AFFINITY_PATH, exchange -> {
+        // add in a handler for servicing affinity requests from Wildfly HTTP client
+        pathHandler.addPrefixPath(COMMON_PATH, getAffinityServiceHandler());
+     }
+
+    private HttpHandler getAffinityServiceHandler() {
+        PathHandler wrappedHandler = new PathHandler();
+        wrappedHandler.addPrefixPath(AFFINITY_PATH, exchange -> {
+            SecureRandomSessionIdGenerator generator = new SecureRandomSessionIdGenerator();
```

Comment:
@rachmatowicz This creates a SecureRandomSessionIdGenerator (and thus a SecureRandom) for each request, which would not be good for performance.